package cms_8120;
option java_package = "com.hikvision.finance.fms.cms.hpp.csclient";
option java_outer_classname = "CsClientRecordProto";
option java_generic_services = true;

//业务单信息表
message CSClientInput
{
	optional int32  id                = 1;         // 业务单Id(更新用)
	optional int32  inputRegionId     = 2;         // 理财系统建单区域号
	optional int32  inputUser         = 3;         // 理财系统建单用户Id
	optional string nproId            = 4;         // 业务单号--业务流水号
	optional string inputTime         = 5;         // 关联业务单时间
	optional string businessTime      = 6;         // 业务时间
	optional string name              = 7;         // 客户姓名
	optional string creditCode        = 8;         // 证件号
	optional int32  creditId          = 9;         // 证件类型Id
	optional int32  productId         = 10;        // 产品Id
	optional string strBankNo         = 11;        // 客户银行卡号
	optional string strBuyMoney       = 12;        // 购买金额
	optional string strCreditPhone    = 13;        // 客户电话
}

//录像表
message CSRecords
{
     optional int32  id                 = 1;      // 录像ID(更新用)
	 optional string strRecordName      = 2;      // 录像名
	 optional int32  inputUser          = 3;      // 理财系统建单用户Id
	 optional int32  inputRegionId      = 4;      // 理财系统建单区域号
	 optional int32  nrecordFileType    = 5;      // 录像文件类型
	 optional string startTime          = 6;      // 录像开始时间
	 optional string endTime            = 7;      // 录像结束时间
	 optional string strUrl             = 8;      // USB本地文件路径
	 optional int32  nupLoadPercent     = 9;      // 记录上传百分比
	 optional int32  nfileSize          = 10;     // 录像大小 
	 optional string screenshotSrc      = 11;     // 截图路径
	 optional string strCloudIp         = 12;     // 云存储-对象存储IP
	 optional int32  strCloudPort       = 13;     // 云存储-对象存储端口
	 optional string strAccessKey       = 14;     // 云存储-对象存储AccessKey
	 optional string strSecretKey       = 15;     // 云存储-对象存储SecretKey
	 optional string strBucket          = 16;     // 云存储-对象存储Bucket

}


message ReqCsClientRecord
{
	optional int32 cmd                     = 1;    // CMD_WDT_CLIENT_RECORD(插入), CMD_MDT_CLIENT_RECORD(更新)
    optional CSClientInput clientInput     = 2;    // 业务单信息表
    repeated CSRecords records             = 3;    // 录像信息表
}

message RspCsClientRecord
{
	optional int32 cmd                     = 1;    // CMD_WDT_CLIENT_RECORD(插入), CMD_MDT_CLIENT_RECORD(更新), CMD_MDT_ERROR_RECORD, CMD_DDT_ERROR_RECORD
    optional int32 result                  = 2;    // 验证结果，0-验证通过，其它值验证失败
    optional string result_info            = 3;    // 信息
}


message RecordsDetailInfo
{
	 optional string  fileName          = 1;     // 录像名
	 optional int32   recordStatus      = 2;     // 录像上传状态Id
	 optional string  strRecordStatus   = 3;     // 录像上传状态
	 optional string  errorMsg          = 4;     // 上传失败原因
	 optional string  strUrl            = 5;     // 录像本地路径
	 optional int32   videoId           = 6;     // 录像Id(关联表prosaverecord主键)
}

message RecordsInfo
{
     optional int32   id                             = 1;      // 业务单ID
	 optional string  nProId                         = 2;      // 业务单号
	 optional int32   inputRegionId                  = 3;      // 区域id
	 optional string  inputRegionCode                = 4;      // 交易区域编码
	 optional string  inputRegionName                = 5;      // 区域名称
	 optional string  custName                       = 6;      // 客户姓名
	 optional string  strCreditPhone                 = 7;      // 客户电话 
	 optional int32   creditId                       = 8;      // 客户证件类型id
	 optional string  creditName                     = 9;      // 证件名称
	 optional string  creditCode                     = 10;      // 客户证件号
	 optional int32   productTypeId                  = 11;      // 产品类型
	 optional string  productTypeName				 = 12;	    // 产品类型名称		
	 optional int32   productId                      = 13;      // 产品id
	 optional string  productCode                    = 14;      // 产品编码
	 optional string  productName                    = 15;      // 产品名称
	 optional string  strBuyMoney                    = 16;      // 购买金额
	 optional string  strGrade						 = 17;      // 产品发行机构
	 optional int32   videoNum                       = 18;      // 录像数量
	 optional string  businessTime                   = 19;      // 业务办理时间
	 optional int32   lockClient                     = 20;      // 业务审批状态Id
	 optional string  strlockClient                  = 21;      // 业务审批状态
	 optional int32   recordStatus                   = 22;      // 录像状态Id
	 optional string  strRecordStatus                = 23;      // 录像状态
	 optional string  pictureStrart                  = 24;      // 左侧图片(没有图片传入[nopic])
	 optional string  pictureEnd                     = 25;      // 右侧图片(没有图片传入[nopic])
	 repeated RecordsDetailInfo recordsDetailInfo    = 26;      // 录像的详细信息
}

//业务单信息请求信息
message ReqRecordsInfo
{
     optional int32  cmd                   = 1;   // CMD_DT_CLIENT_RECORD(查询), CMD_DT_ERROR_RECORD(查询)
     optional int32  recordTab             = 2;   // 保留字段
     optional string startDate             = 3;   // 开始时间过滤条件
     optional string endDate               = 4;   // 结束时间过滤条件
     optional int32  pageNo                = 5;   // 当前页数
     optional int32  pageSize              = 6;   // 当前查询的数据量 default[10] 
     optional int32  inputUser             = 7;   // 用户id
     optional string productName           = 8;   // 产品名称
     optional string custName              = 9;   // 客户姓名
}

//业务单信息返回信息
message RspCsClientRecordsInfo
{
	optional int32 cmd                     = 1;    // CMD_DT_CLIENT_RECORD(查询)
    optional int32 result                  = 2;    // 验证结果，0-验证通过，其它值验证失败
    optional string result_info            = 3;    // 信息
    repeated RecordsInfo recordsInfo       = 4; 
}

//录像异常信息
message ErrRecordsInfo 
{
	optional int32   id                     = 1;    //录像Id(录像关联表中的主键) 
	optional string  fileName               = 2;    //录像名
	optional string  nProId                 = 3;    //业务单号
	optional string  custName               = 4;    //客户姓名
	optional string  businessTime           = 5;    //办理时间
	optional int32   recordStatus           = 6;    //录像状态Id
	optional string  strRecordStatus        = 7;    //录像状态
	optional int32   productTypeId          = 8;   // 产品类型
	optional string  productTypeName	    = 9;   // 产品类型名称		
	optional int32   productId              = 10;   // 产品id
	optional string  productCode            = 11;   // 产品编码
	optional string  productName            = 12;   // 产品名称
	optional string  strBuyMoney            = 13;   // 购买金额
	optional int32   recordCheck            = 14;   // 合规状态Id
	optional string  strRecordCheck         = 15;   // 合规状态
	optional string  strUrl                 = 16;   // 录像路径
}

//录像异常返回信息
message RspErrRecordsInfo
{
	optional int32 cmd                      = 1;    // CMD_DT_ERROR_RECORD(查询)
    optional int32 result                   = 2;    // 验证结果，0-验证通过，其它值验证失败
    optional string result_info             = 3;    // 信息
    repeated ErrRecordsInfo errRecordsInfo  = 4; 
}

//异常录像请求
message ReqErrorRecord
{
	optional int32 cmd                     = 1;    // CMD_DDT_CLIENT_RECORD(删除)
	repeated int32 id                      = 2;    // 录像id
	optional int32 inputUser               = 3;    // 用户id
}

//打印业务单请求
message ReqPrintRecord
{
	optional int32 cmd                     = 1;    // CMD_WDT_CLIENT_RECORD(插入), CMD_MDT_CLIENT_RECORD(更新)
	optional int32 id                      = 2;    // 业务单id
	optional int32 inputUser               = 3;    // 用户id
}


//返回打印业务单信息
message RspPrintRecord
{
	optional int32 cmd                         = 1;    // CMD_WDT_CLIENT_RECORD(插入), CMD_MDT_CLIENT_RECORD(更新), CMD_MDT_ERROR_RECORD, CMD_DDT_ERROR_RECORD
    optional int32 result                      = 2;    // 验证结果，0-验证通过，其它值验证失败
    optional string result_info                = 3;    // 信息
    optional int32 id                          = 4;    // 业务单id
	optional string name                       = 5;    // 客户姓名
	optional string creditName                 = 6;    // 证件名称
	optional string creditCode                 = 7;    // 证件编码
	optional string nproId                     = 8;    // 业务流水号
	optional string strGrade                   = 9;    // 发行机构
	optional string productType                = 10;    // 产品类型名称
	optional string productName                = 11;    // 产品名称
	optional string proName                    = 12;    // 业务员姓名
	optional string proUserNo                  = 13;   // 业务员编号
	optional string pictureStrart              = 14;   // 左侧图片(没有图片传入[nopic])
	optional string pictureEnd                 = 15;   // 右侧图片(没有图片传入[nopic])
	optional string businessTime               = 16;   // 交易时间
}

service ModifyCsClientRecordService {
	
	rpc SaveCsClientRecord(ReqCsClientRecord) returns (RspCsClientRecord);	         // CMD_WDT_CLIENT_RECORD(插入业务单信息)
	rpc UpdateCsClientRecord(ReqCsClientRecord) returns (RspCsClientRecord);          // CMD_MDT_CLIENT_RECORD(更新业务单信息)
	rpc QueryCsClientRecords(ReqRecordsInfo) returns (RspCsClientRecordsInfo);       // CMD_DT_CLIENT_RECORD(查询业务单信息)
	rpc DeleteErrorRecords(ReqErrorRecord) returns (RspCsClientRecord);              // CMD_DDT_ERROR_RECORD(删除异常录像信息)
	rpc QueryErrorRecords(ReqRecordsInfo) returns (RspErrRecordsInfo);               // CMD_DT_ERROR_RECORD(查询异常录像信息)
	rpc UploadErrorRecords(ReqErrorRecord) returns (RspCsClientRecord);              // CMD_MDT_ERROR_RECORD(重新上传异常录像)
	rpc PrintCsClientRecords(ReqPrintRecord) returns (RspPrintRecord);              // CMD_DT_PRINT_RECORD(打印业务单信息)
}

